src/components/UserSelector.tsx

```diff
@@ -1,16 +1,77 @@
-import React from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { User } from '../types/User';
+import cn from 'classnames';
+import { getUsers } from '../api/users';
+
+type Props = {
+  selectedUser: User | null;
+  onSelectUser: (user: User) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  selectedUser,
+  onSelectUser,
+}) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [users, setUsers] = useState<User[]>([]);
+  const [error, setError] = useState(false);
+
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  const loadUsers = async () => {
+    try {
+      const usersFromServer = await getUsers();
+
+      setUsers(usersFromServer);
+    } catch {
+      setError(true);
```

Comment:
It's a good practice to provide an error message in the catch clause, so that you could debug the application more effectively. You could consider logging the error or displaying it to the user instead of just setting the error state to true.